### ISSUE TITLE
Enhancement: [not-under-rootdir] rule to ignore TS errors involving using code that is not inside the rootDir

### ISSUE THREAD

**Issue by jcollum-nutrien** [typescript-eslint/typescript-eslint#5081]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/

### Description

I don't understand how "Link to the rule's documentation" is compatible with "Propose a new lint rule option" here. 

The "is not under 'rootDir'" rule should be switchable. 

https://github.com/microsoft/TypeScript/blob/0414deeaaf211ee000548c30fd81aa09b311abb7/src/compiler/diagnosticMessages.json#L4330

tsconfig.json: 

```
 "rootDir": "./src",
```

Actual error: `TS6059: File '/Users/justin.collum2@nutrien.com/work/seed-association-service/data/test/adjustRankings/714-rankings.json' is not under 'rootDir' '/Users/justin.collum2@nutrien.com/work/seed-association-service/src'. 'rootDir' is expected to contain all source files.`

In my use case I have some json files in /data and a test in /src. Since the files are json there's no compilation. If I add `@ts-ignore` to those lines the code runs fine. 

Maybe this should be a "allow files outside of rootDir but only if they are not Typescript files"? 

### Fail

```typescript
// in the src dir, in a test file: 
import allocations714 from '../data/test/adjustRankings/714-allocations.json';
```

### Pass

```typescript
// literally the same
import allocations714 from '../data/test/adjustRankings/714-allocations.json';
```

### Additional Info

NBD if this is not possible.

**Comment by JoshuaKGoldberg** [comment]
@jcollum-nutrien you're in the wrong repository, friend 🙂. TypeScript ESLint is a set of tooling that enables ESLint to support TypeScript. You want the repository for TypeScript, https://github.com/Microsoft/TypeScript.

See explanations on https://typescript-eslint.io/.
